### PR TITLE
Fix bug not showing dappIcon

### PR DESCRIPTION
**Pull request by amaury1093**
- Fix bug of dappIcons not showing
- Remove imageRef when hiding broken images, use CSS class instead

### CODE REVIEW

File: src/DappIcon/dappIcon.js [hunk 1]
```diff
@@ -14,64 +14,81 @@
...
 
-    const fallbackSrc = window.location.protocol === 'file:'
-      ? `dapps/${app.id}/icon.png`
-      : `${dapphost}/dapps/${app.id}/icon.png`;
+    let imageSrc;
+
+    switch (app.type) {
+      case 'view': {
+        const fallbackSrc =
+          window.location.protocol === 'file:' ? `dapps/${app.id}/icon.png` : `${dappHost}/dapps/${app.id}/icon.png`;
```
Comment (amaury1093) [inline src/DappIcon/dappIcon.js]: @jacogr Can you confirm that we are not using file:// anymore?

In that case I'll just put `fallbackSrc = ${dappHost}/dapps/${app.id}/icon.png`

File: src/DappIcon/dappIcon.js [hunk 2]
```diff
@@ -14,64 +14,81 @@
...
+        imageSrc = app.image ? `${dappHost}${app.image}` : fallbackSrc;
+        break;
+      }
+      case 'local':
+        imageSrc = `${this.dappsUrlStore.dappsUrl}/${app.id}/${app.iconUrl}`;
+        break;
+      case 'builtin':
+      case 'network':
+      default:
+        imageSrc = app.image; // TODO: should be `${UI_URL}${app.image}` where UI_URL=http://localhost:8180
```
Comment (amaury1093) [inline src/DappIcon/dappIcon.js]: I see parity_dappsUrl and parity_wsUrl, but no parity_uiUrl. I left it like this (app.image is something like `/api/content/{hash}/icon.png`), and it works fine. But the ideal scenario would be to use UI_URL given by the API.

File: src/DappIcon/dappIcon.js (hunk 1 above)
Comment (jacogr) [reply src/DappIcon/dappIcon.js]: We are using it for Electron. (WIP, will become important going forward)

File: src/DappIcon/dappIcon.js (hunk 2 above)
Comment (jacogr) [reply src/DappIcon/dappIcon.js]: UI url is always window.location.host.

Comment (amaury1093) [reply src/DappIcon/dappIcon.js]: In a dapp, window.location.host give localhost:3001, and window.parent.location.host is inaccessible

File: src/DappIcon/dappIcon.js (hunk 1 above)
Comment (amaury1093) [reply src/DappIcon/dappIcon.js]: OK so I'll leave it like this

**Pull request merged**